ircd/chmode.c

```diff
@@ -1538,8 +1538,13 @@ set_channel_mode(struct Client *client_p, struct Client *source_p,
 				*mbuf = '\0';
 
 				if(cur_len > mlen)
+				{
 					sendto_channel_local_priv(IsServer(source_p) ? fakesource_p : source_p,
 							send_flags, priv, chptr, "%s %s", modebuf, parabuf);
+					if (priv != NULL && IsClient(source_p))
```

Comment:
This doesn't appear to be the test you think it is? `priv` is only set to non-null when `flags` is `ONLY_OPERS` (in which case it is set to `auspex:cmodes`, which is not the priv used to set the mode), and the only mode changes with `mems` set to `ONLY_OPERS` are `chm_hidden` modes.